d2/src/main/java/com/linkedin/d2/balancer/dualread/DualReadLoadBalancer.java

```diff
@@ -212,13 +220,14 @@ public void getLoadBalancedServiceProperties(String serviceName, Callback<Servic
         _newLb.getLoadBalancedServiceProperties(serviceName, clientCallback);
         break;
       case DUAL_READ:
-        if (_newLbExecutor.isShutdown())
+        try
         {
-          _rateLimitedLogger.info("newLb executor is shutdown already. Skipping getLoadBalancedServiceProperties on newLb executor.");
+          _newLbExecutor.execute(() -> _newLb.getLoadBalancedServiceProperties(serviceName, Callbacks.empty()));
         }
-        else
+        catch (RejectedExecutionException e)
         {
-          _newLbExecutor.execute(() -> _newLb.getLoadBalancedServiceProperties(serviceName, Callbacks.empty()));
+          _rateLimitedLogger.debug("newLb executor rejected new task for getLoadBalancedServiceProperties. "
```

Comment:
the "other executor inside newLb executor" is guarded separately in SimpleLoadBalancer.
Yes, with the reject handler in container, currently newLb executor won't throw exceptions when rejecting tasks. Just adding this guard in general in case the container code gets changed in the future (by mistake, etc.).